### DISCUSSION TITLE
How about support new target as plugin

### DISCUSSION THREAD

**Discussion by cyqw** [antlr/antlr4#4743] [Ideas]
If someone want to create a new target for a language, currently have to add all the code of the target ('codegen', 'runtime' and 'test') to this repository.
I think the new target should be stored in a separate repository. It can be maintained and released independently.
The 'codegen' of the new target need depends on antlr's 'tools'. It should base on the big version of antlr.
The 'codegen' of the new target is a plugin of antlr, antlr searches the plugin folder to find the new target, then it can generate the lexer parser code of that target.

**Comment by ericvergnaud** [comment]
Thanks. Antlr5 is designed that way.

**Reply by kaby76** [reply]
Antlr5 hasn't had any meaningful PRs since March 20. What's the status?

**Comment by ericvergnaud** [comment]
It’s pending availability of its contributors
